### ISSUE TITLE
OpenAPI (Swagger) schema shouldn't provide choices for Status references

### ISSUE THREAD

**Issue by glennmatthews** [nautobot/nautobot#3215]
### Environment
* Nautobot version (Docker tag too if applicable): all

### Steps to Reproduce

Currently the OpenAPI schema generated by Nautobot will list a specific enum of permissible options for references to Status objects, e.g.

```yaml
    Circuit:
      ...
      properties:
        ...
        status:
          type: object
          properties:
            value:
              type: string
              enum:
              - active
              - decommissioned
              - deprovisioning
              - offline
              - planned
              - provisioning
            label:
              type: string
              enum:
              - Active
              - Decommissioned
              - Deprovisioning
              - Offline
              - Planned
              - Provisioning
```

The problem, of course, is that this is NOT actually a hard-coded list of choices, it's determined at schema generation time from whatever set of Status objects happen to be defined. The moment a user adds, deletes, or edits any Status record, this part of the schema becomes inaccurate.

Most likely, the best solution here (albeit probably a breaking, i.e. 2.0 change) will be to change the `status` API field serializers so that they use a nested serializer like any other foreign-key reference, i.e. instead of:

```json
      "status": {
        "value": "active",
        "label": "Active"
      }
```

they instead use:

```json
    {
      "id": "64864569-9d3f-4b17-bb00-47296d9ccd0e",
      "display": "Active",
      "url": "https://demo.nautobot.com/api/extras/statuses/64864569-9d3f-4b17-bb00-47296d9ccd0e/",
      "name": "Active",
      "slug": "active",
    },
```

**Comment by glennmatthews** [comment]
Straightforward enough to implement, thanks to extensive use of mixin classes. The bigger challenge will be making it backwards-compatible with older API versions - there are a lot of models with Statuses, so that'll be a lot of versioned serializers.

**Comment by glennmatthews** [comment]
Implemented for 2.0 in #3225.
